taf/updater/updater.py

```diff
@@ -425,6 +447,18 @@ def _update_named_repository(
     if url in visited:
         return
     visited.append(url)
+
+    # JMC: Added logic for no_upstream
+    if no_upstream:
```

Comment:
This is probably not what we want to do here, please take a look at the previous comment. The issue lacked some important information.